src/internal/_xtake.js

```diff
@@ -15,7 +15,8 @@ module.exports = (function() {
       return _reduced(result);
     } else {
       this.n -= 1;
-      return this.xf['@@transducer/step'](result, input);
+      return this.n === 0 ? _reduced(this.xf['@@transducer/step'](result, input))
+                          : this.xf['@@transducer/step'](result, input);
     }
```

Comment:
I don't think so, but I'm not familiar with the codebase. 

I've tried a couple of things without success.
Adding some kind of init flag might make it less clear.

What do you say? Should we keep it like this?